BoxContentSDK/BoxContentSDK/Requests/BOXFolderItemsRequest.m

```diff
@@ -80,58 +83,79 @@ + (NSArray *)dedupeItemsByBoxID:(NSArray *)items
 
 - (void)performRequestWithCompletion:(BOXItemsBlock)completionBlock
 {
+    [self performRequestWithCached:nil refreshed:completionBlock];
+}
+
+- (void)performRequestWithCached:(BOXItemsBlock)cacheBlock refreshed:(BOXItemsBlock)refreshBlock
+{
     __weak BOXFolderItemsRequest *weakSelf = self;
-    BOOL isMainThread = [NSThread isMainThread];
     
-    NSMutableArray *results = [[NSMutableArray alloc] init];
-    
-    // used to work around warning on retain loop on paginatedItemFetch
-    __block BOXItemArrayCompletionBlock recursiveFetch = nil;
-    
-    BOXItemsBlock localCompletionBlock = ^(NSArray *items, NSError *error){
-        [BOXDispatchHelper callCompletionBlock:^{
-            completionBlock(items, error);
-        } onMainThread:isMainThread];
-    };
+    if (cacheBlock) {
+        if (self.requestCache) {
```

Comment:
If requestCache is nil, should we just surface up a nil object in cacheBlock? Otherwise, the caller might expect the completion block to be called.